extensions/github1s/src/listeners/vscode.ts

```diff
@@ -31,13 +31,14 @@ export const registerVSCodeEventListeners = () => {
 			return;
 		}
 
-		if (await shouldClosedThisEditor(editor)) {
-			vscode.commands.executeCommand(
-				'workbench.action.closeActiveEditor',
-				activeFileUri
-			);
-			return;
-		}
+		// TODO: How to deal with opened editor?
+		// if (await shouldClosedThisEditor(editor)) {
+		// 	vscode.commands.executeCommand(
+		// 		'workbench.action.closeActiveEditor',
+		// 		activeFileUri
+		// 	);
+		// 	return;
+		// }
```

Comment:
Is it possible the save and restore the workspace based on active authority? So that, if the authority changes, it will create a new workspace or restore from the cache.

Reply:
Thank you so much! It seems not easy. The code below generate the uniqueId of the workspace, but we can't access the Extension State right now.
https://github.com/conwnet/github1s/blob/ada39590e276703b48027dc21e3611752f324ea6/src/vs/workbench/browser/web.main.ts#L365-L374

And create a new workspace is not easy too, I've tried that. We use the single-folder workspace now, update the workspace will create an new multi-folder worspace or refresh the browser.
I've been trying to figure it out how to replace current workspace without refresh browsers.😂

Reply:
Thanks for the explanation, it's just an ambitious idea and we could improve the experience in the future.

> how to replace current workspace without refresh browsers.

This is exactly what we need so that to delegate the UI layout work to VS Code.